ext/AdvancedHMCADTypesExt.jl

```diff
@@ -0,0 +1,41 @@
+module AdvancedHMCADTypesExt
+
+if isdefined(Base, :get_extension)
+    using AdvancedHMC:
+        AbstractMetric,
+        Hamiltonian,
+        LogDensityModel,
```

Comment:
It would be cleaner to directly import it from AbstractMCMC instead of relying on it being imported in AdvancedHMC.